src/common/utility.js

```diff
@@ -88,6 +88,8 @@ class PromiseClass {
     }
 }
 
+const isEven = (index, reverse) => (reverse ? (index + 1) % 2 : index % 2)
```

Comment:
Minor suggestion to rename it to `isIndexEven` (for clarity). Nice refactor

Reply:
Ah sweet, was struggling to come up with a concise & meaningful name. Thanks 👍